addons/mail/static/src/models/thread_view/thread_view.js

```diff
@@ -103,6 +103,22 @@ registerModel({
             const { length, [length - 1]: lastMessageView } = this.messageViews;
             return lastMessageView ? replace(lastMessageView) : clear();
         },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeMessageListView() {
+            if (!this.thread || !this.threadCache) {
+                return clear();
+            }
+            if (this.isLoading && !this.threadCache.isLoaded) {
+                return clear();
+            }
+            if (this.threadCache.isLoaded || this.thread.isTemporary) {
+                return insertAndReplace();
+            }
+            return clear();
```

Comment:
strange to have made the complex condition, but not made the element actually "elif" independent in the template, maybe go all the way and also adapt the other conditions in template?

Reply:
I was gonna do this, but just saw below another `t-elif`, so this looks like a spool of thread if I start doing it.

Reply:
Personally I don't like `t-elif` in template, so goal would be to remove all `t-elif` that are intertwined.
It's more delicate than it looks, as we need to check all combinations of state with UI to make sure nothing is broken by change. This may require introduce new concepts so that the boolean expressions are properly specific to make it clear these parts of template are mutually exclusive.